catalog/tests/dags/database/staging_database_restore/test_staging_database_restore.py

```diff
@@ -109,3 +115,45 @@ def test_make_rename_task_group():
     # This should not be affected by the trigger rule
     assert await_rename.trigger_rule == TriggerRule.ALL_SUCCESS
     assert await_rename.retries == 2
+
+
+@pytest.mark.parametrize(
+    "tags, expected",
+    [
+        (
+            ["11926a78cfe8ca150fe6d232a5689141c35417d1", "latest"],
+            "11926a78cfe8ca150fe6d232a5689141c35417d1",
+        ),
+        (
+            ["rel-2023.05.30.17.57.04", "latest"],
+            "rel-2023.05.30.17.57.04",
+        ),
+        (
+            ["rel-2023.05.30.17.57.04"],
+            "rel-2023.05.30.17.57.04",
+        ),
+        (
+            [
+                "11926a78cfe8ca150fe6d232a5689141c35417d1",
+                "2e82df0105a94f7b85c48321fc3e05c23edc49bd",
+                "latest",
+            ],
+            "11926a78cfe8ca150fe6d232a5689141c35417d1",
+        ),
+        pytest.param(
+            ["11926a78cfe8ca150fe6d232a5689141c35417d1"],
+            None,
+            marks=pytest.mark.raises(
+                exception=ValueError, match="Latest version is not"
+            ),
+        ),
```

Comment:
So this case will fail because it's not a "rel-" kind of tag and doesn't have the "latest" either, gotcha!